build.gradle

```diff
@@ -47,3 +57,99 @@ tasks.named("dependencyUpdates").configure {
         isNonStable(it.candidate.version)
     }
 }
+
+/**
+ * Compares the current project dependencies with the ones stored in the dependencies.lock file
+ * Fails if the dependencies do no match and outputs the difference
+ */
+task compareDependencies {
```

Comment:
@alpeto9 When do we need to run this task? Did you define it in case we update the dependencies?

Reply:
It needs to be run the following command: `$ ./gradlew generateDependenciesLockFile ` 
Task `compareDependencies` is being called by `generateDependenciesLockFile` @micbakos-rdx

Reply:
I ran the command locally in this branch and got these changes. What are these supposed to do?
Actually why should we have changes here?
<img width="1783" alt="Screenshot 2024-10-23 at 17 53 25" src="https://github.com/user-attachments/assets/19e2c6e6-6000-4b85-ad48-a4ceaf945335">

Reply:
or before you merge you have to run the `generateDependenciesLockFile` and the `compareDependencies` ?

Reply:
To which file are you comparing to?

Reply:
Before merging run the command `$ ./gradlew generateDependenciesLockFile` which generates `dependencies.lock` file.

My `dependencies.lock` file is slightly different as yours, since my local setup is slightly different, but I've compared @giannis-rdx and @micbakos-rdx generated dependencies.lock file and they are identical.

Reply:
Just for the record I would suggest running Java 17.